File: comandor/src/main/java/com/komandor/komandor/data/database/certificates/CertificateDao.java
package com.komandor.komandor.data.database.certificates;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import io.reactivex.Flowable;

@Dao
public interface CertificateDao {
  
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  void insert(Certificate certificate);
  
  @Query("select * from " + Certificate.TABLE_NAME + " where is_active = 1")
  Flowable<Certificate> getActiveCertificate();
  
  @Query("select * from " + Certificate.TABLE_NAME + " where is_local = 1")
  Flowable<List<Certificate>> getLocalCertificates();

  @Query("select * from " + Certificate.TABLE_NAME + " where cert_id = :cid")
  Flowable<Certificate> getCertificateByID(int cid);

  @Query("select base64 from " + Certificate.TABLE_NAME + " where cert_id = :cid")
  String getCertificateStringByIDSync(int cid);
  
  @Query("select * from " + Certificate.TABLE_NAME + " where contact_id = :contactID")
  Flowable<List<Certificate>> getCertificatesByContactID(int contactID);
}

File: comandor/src/main/java/com/komandor/komandor/data/database/dao/CertificateDao.kt
package com.komandor.komandor.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.komandor.komandor.data.database.model.Certificate
import kotlinx.coroutines.flow.Flow

@Dao
abstract class CertificateDao: BaseDao<Certificate>() {

    @Query("select * from CERTIFICATES where is_active = 1")
    abstract fun getActiveCertificate(): Flow<Certificate?>?

    @Query("select * from CERTIFICATES where is_local = 1")
    abstract fun getLocalCertificates(): Flow<List<Certificate?>?>?

    @Query("select * from CERTIFICATES where cert_id = :cid")
    abstract fun getCertificateByID(cid: Int): Flow<Certificate?>?

    @Query("select base64 from CERTIFICATES where cert_id = :cid")
    abstract fun getCertificateStringByIDSync(cid: Int): String?

    @Query("select * from CERTIFICATES where contact_id = :contactID")
    abstract fun getCertificatesByContactID(contactID: Int): Flow<List<Certificate?>?>?
}